dis_snek/models/discord_objects/scheduled_event.py

```diff
@@ -0,0 +1,204 @@
+
+@define()
+class ScheduledEventUser(DiscordObject):
+    user: "User" = attr.ib()
+    member: "Member" = attr.ib(default=MISSING)
+
+    @classmethod
+    def _process_dict(cls, data: Dict[str, Any], client: "Snake") -> Dict[str, Any]:
+        if data.get("member"):
+            fixed_member = data
+            fixed_member["user"]["member"] = fixed_member["member"]
+            data["member"] = client.cache.place_member_data(data["guild_scheduled_event_id"], fixed_member["user"])
+        data["user"] = client.cache.place_user_data(data["user"])
+        data["id"] = data["user"].id
+        return data
+
+
+@define()
+class ScheduledEvent(DiscordObject):
+    name: str = attr.ib()
+    description: str = attr.ib(default=MISSING)
+    entity_type: Union[ScheduledEventEntityType, int] = attr.ib(converter=ScheduledEventEntityType)
+    """	the type of the scheduled event"""
+    scheduled_start_time: Timestamp = attr.ib(converter=timestamp_converter)
+    """ a Timestamp object representing the scheduled start time of the event """
+    scheduled_end_time: Optional[Timestamp] = attr.ib(default=MISSING, converter=optional(timestamp_converter))
+    """	optinal Timstamp object representing the scheduled end time, required if entity_type is EXTERNAL"""
+    privacy_level: Union[ScheduledEventPrivacyLevel, int] = attr.ib(converter=ScheduledEventPrivacyLevel)
+    """the privacy level of the scheduled event"""
+    status: Union[ScheduledEventStatus, int] = attr.ib(converter=ScheduledEventStatus)
+    """	the status of the scheduled event"""
+    entity_id: Optional["Snowflake_Type"] = attr.ib(default=MISSING, converter=optional(to_snowflake))
+    """the id of an entity associated with a guild scheduled event"""
+    entity_metadata: Optional[Dict[str, Any]] = attr.ib(default=MISSING)  # TODO make this
+    """the metadata associated with the entity_type"""
+    user_count: int = attr.ib(default=MISSING)
+    """	the number of users subscribed to the scheduled event"""
+
+    _guild_id: "Snowflake_Type" = attr.ib(converter=to_snowflake)
+    _creator: Optional["User"] = attr.ib(default=MISSING)
+    _creator_id: Optional["Snowflake_Type"] = attr.ib(default=MISSING, converter=optional(to_snowflake))
+    _channel_id: Optional["Snowflake_Type"] = attr.ib(default=None, converter=optional(to_snowflake))
+
+    @property
+    async def creator(self) -> Optional["User"]:
+        """Returns the user who created this event
+
+        !!! note:
+            Events made before October 25th, 2021 will not have a creator.
+        """
+        return await self._client.cache.get_user(self._creator_id) if self._creator_id else None
+
+    @property
+    def guild(self) -> "Guild":
+        return self._client.cache.guild_cache.get(self._guild_id)
+
+    @classmethod
+    def _process_dict(cls, data: Dict[str, Any], client: "Snake") -> Dict[str, Any]:
+        if data.get("creator"):
+            data["creator"] = client.cache.place_user_data(data["creator"])
```

Comment:
`creator` is a user in the api spec, but you could *attempt* to get a member object from the cache as this **will** occur in a guild

Reply:
this is to help out wolfhound here: is that a request or a thought you jotted down?

Reply:
A request